### PR TITLE
JDK-8301201: Allow \n@ inside inline tags using inlineContent

### PR DESCRIPTION
**Pull request by jonathan-gibbons**
Please review a simple change to allow the use of _newline_ _whitespace_ `@` inside inline tags that allow rich content (that is, those parsed with `inlineContent`) as compared to those that only allow plain text (that is, those parsed with `inlineText`).

The fix is to delete the code which recognizes `@` as the beginning of a block tag. Compare to the similar fix in [JDK-8241780](https://bugs.openjdk.org/browse/JDK-8241780)

The general `TagTest.java` is updated for the new feature.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8301201](https://bugs.openjdk.org/browse/JDK-8301201): Allow \n@ inside inline tags using inlineContent

### Reviewers
 * [Hannes Wallnöfer](https://openjdk.org/census#hannesw) (@hns - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk pull/12264/head:pull/12264` \
`$ git checkout pull/12264`

Update a local copy of the PR: \
`$ git checkout pull/12264` \
`$ git pull https://git.openjdk.org/jdk pull/12264/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 12264`

View PR using the GUI difftool: \
`$ git pr show -t 12264`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/12264.diff">https://git.openjdk.org/jdk/pull/12264.diff</a>

</details>

### CODE REVIEW

File: test/langtools/tools/javac/doctree/DocCommentTester.java [hunk 1]
```diff
@@ -919,7 +919,7 @@ void check(TreePath path, Name name) throws Exception {
         String normalize(String s) {
             String s2 = s.trim().replaceFirst("\\.\\s*\\n *@", ".\n@");
             StringBuilder sb = new StringBuilder();
-            Pattern p = Pattern.compile("\\{@(code|literal)( )?");
+            Pattern p = Pattern.compile("(?i)\\{@([a-z][a-z0-9.:-]*)( )?");
```
Comment (hns) [inline test/langtools/tools/javac/doctree/DocCommentTester.java:922]: Why is this change necessary? I see it is intended to impact the unknown inline tag in `TagTest.java`, but couldn't the test be written without this change?

**Comment by hns** [comment]
There's something wrong with this, several `javac/doctree` tests are failing.

File: test/langtools/tools/javac/doctree/DocCommentTester.java (hunk 1 above)
Comment (jonathan-gibbons) [reply test/langtools/tools/javac/doctree/DocCommentTester.java:934]: Previously, the only tags in which `newline at` was supported were `{@code}` `{@literal}` and  `{@snippet}` ... and this tag predated `{@snippet}` which is the main reason it's not included here.

Now `newline at` is supported in all inline tags, so the scope of this check is broadened accordingly.

**Comment by jonathan-gibbons** [comment]
> There's something wrong with this, several `javac/doctree` tests are failing.

Yes, now fixed. The breakage was obviously due to the change you mentioned above (for `code|literal` tags). The fix in most cases (one exception) was to make the `PrettyChecker` more robust. The one difficult case was a syntactically invalid tag (which was in most ways a reasonable test case) but which led to false positive matches in the `normalize` code. For that case, I created a "backdoor" to pass a boolean flag to the `normalize` code by means of an annotation on the test case.  Note that because the test case is only _parsed_ and never _analyzed_, the annotation does not need a declaration, it's enough to have an annotation show up in the AST, where it can be accessed by the `PrettyChecker` code.

**Review by hns — APPROVED** [review]
Thanks for the extended comment for the `normalize` method. Looks good to me.

**Comment by jonathan-gibbons** [comment]
@hns Thanks for the reviews.
